src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -387,13 +387,7 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $lowercase = true)
                         $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE;
                     }
 
-                    if ($strict = $arg->getAttribute('strict')) {
```

Comment:
I would still trigger a deprecation if the attribute is there, otherwise we cannot remove it from the XSD in 4.x